Sources/Plasma/CoreLib/hsWindows.h

```diff
@@ -97,12 +97,17 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
     // Initializes COM exactly once the first time it's called.
     void hsRequireCOM();
 
+    /** Represents the last Win32 error. */
+    struct hsLastWin32Error_Type {};
+    constexpr hsLastWin32Error_Type hsLastWin32Error;
+
     /** COM Result holder used for formatting to log. */
     struct hsCOMError
     {
         HRESULT fResult;
 
         hsCOMError() : fResult() { }
+        hsCOMError(hsLastWin32Error_Type) : fResult(HRESULT_FROM_WIN32(GetLastError())) { }
```

Comment:
This must be used carefully...  For example, if you construct an `hsCOMError()` as a parameter to a function call (or constructor), you need to make sure none of the other parameters will set the last error and cause this to get the wrong value...

Personally, I would prefer to see `GetLastError()` called explicitly by the calling code, and then an `hsCOMError` constructed with the result (presumably still with a tag type to distinguish it from the `HRESULT` overload)